src/main/java/org/commonwl/view/git/GitDetails.java

```diff
@@ -20,18 +20,51 @@
 package org.commonwl.view.git;
 
 import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.web.client.RestTemplate;
+import org.springframework.web.util.UriComponents;
+import org.springframework.web.util.UriComponentsBuilder;
 
 import java.io.Serializable;
 import java.net.URI;
 import java.net.URISyntaxException;
+import java.net.URLEncoder;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Map;
 import java.util.Objects;
 
+import static java.util.Map.entry;
+
 /**
  * Represents all the parameters necessary to access a file/directory with Git
  */
 @JsonIgnoreProperties(value = {"internalUrl"}, ignoreUnknown = true)
 public class GitDetails implements Serializable {
 
+    private final Logger logger = LoggerFactory.getLogger(this.getClass());
+
+    private static final String SPDX_PREFIX = "https://spdx.org/licenses/";
+
+    private static final Map<String, String> GITLAB_TO_SPDX_LICENSES = Map.ofEntries(
+            entry("agpl-3.0", "AGPL-3.0-or-later"),
```

Comment:
Is this always true? What is the source of these mappings?

Reply:
I substituted this with the `spdx-license-checker` and using the `"name"` field from the GitLab JSON response